### ISSUE TITLE
Building the ssl extension fails with cmake 3.0

### ISSUE THREAD

**Issue by elemoine** [mozilla-services/lua_sandbox_extensions#19]
With this issue I just want to report that building the ssl extension fails with cmake 3.0.2, which is the version shipped with Debian Jessie.

This is the error I get at `make` time:

```
root@1b2e35c5cfd3:/workdingdir/lua_sandbox_extensions/release# make
Scanning dependencies of target ep_ssl
[ 11%] Creating directories for 'ep_ssl'
[ 22%] Performing download step (git clone) for 'ep_ssl'
-- Avoiding repeated git clone, stamp file is up to date: '/workdingdir/lua_sandbox_extensions/release/ssl/ep_ssl-prefix/src/ep_ssl-stamp/ep_ssl-gitclone-lastrun.txt'
[ 33%] No patch step for 'ep_ssl'
[ 44%] Performing update step for 'ep_ssl'
[ 55%] Performing configure step for 'ep_ssl'
Not searching for unused variables given on the command line.
CMake Error at /usr/share/cmake-3.0/Modules/FindOpenSSL.cmake:293 (list):
  list GET given empty list
Call Stack (most recent call first):
  CMakeLists.txt:41 (find_package)


CMake Error at /usr/share/cmake-3.0/Modules/FindOpenSSL.cmake:294 (list):
  list GET given empty list
Call Stack (most recent call first):
  CMakeLists.txt:41 (find_package)


CMake Error at /usr/share/cmake-3.0/Modules/FindOpenSSL.cmake:296 (list):
  list GET given empty list
Call Stack (most recent call first):
  CMakeLists.txt:41 (find_package)


CMake Error at /usr/share/cmake-3.0/Modules/FindOpenSSL.cmake:298 (list):
  list GET given empty list
Call Stack (most recent call first):
  CMakeLists.txt:41 (find_package)


-- Found OpenSSL: /usr/lib/x86_64-linux-gnu/libssl.so;/usr/lib/x86_64-linux-gnu/libcrypto.so (found version ".0.0`") 
-- Configuring incomplete, errors occurred!
See also "/workdingdir/lua_sandbox_extensions/release/ssl/ep_ssl-prefix/src/ep_ssl-build/CMakeFiles/CMakeOutput.log".
ssl/CMakeFiles/ep_ssl.dir/build.make:103: recipe for target 'ssl/ep_ssl-prefix/src/ep_ssl-stamp/ep_ssl-configure' failed
make[2]: *** [ssl/ep_ssl-prefix/src/ep_ssl-stamp/ep_ssl-configure] Error 1
CMakeFiles/Makefile2:1005: recipe for target 'ssl/CMakeFiles/ep_ssl.dir/all' failed
make[1]: *** [ssl/CMakeFiles/ep_ssl.dir/all] Error 2
Makefile:86: recipe for target 'all' failed
make: *** [all] Error 2
```

It looks like cmake 3.0.2 is not able to determine the version OpenSSL (`found version ".0.0"` does not look correct).

I tried with cmake 3.5.2, downloaded from the cmake website, and it works. So the problem is related to cmake 3.0.2 (and possibly other cmake versions between 3.0.2 and 3.5.2).

I am creating this issue mainly for posterity. One thing that could be done though is to change the required cmake version on the README page.

**Comment by trink** [comment]
Updated the requirements to 3.5

**Comment by elemoine** [comment]
Thanks.

**Comment by sathieu** [comment]
Yes, this looks to be https://cmake.org/gitweb?p=cmake.git;a=commitdiff;h=c5d9a8283cfac15b4a5a07f18d5eb10c1f388505, first in 3.1.2.

**Comment by sathieu** [comment]
And it'll be fixed in upcoming Debian 8.6.

Ref:
- https://bugs.debian.org/826656
- https://lists.debian.org/debian-stable-announce/2016/09/msg00000.html
